Flutter/calendar/appointments.md

```diff
@@ -130,6 +130,129 @@ events.notifyListeners(CalendarDataSourceAction.reset, null);
 {% endhighlight %}
 {% endtabs %}
 
+## Datasource manipulation
+
+By call this method whenever the datasource changes, to notify the datasource may have. Need to notify the datasource action(add, remove, and reset) through notify listener. In the following example, the appointment and resource are added, removed, and replaced and notifyListeners is called in onPressed callback of the FloatingActionButton.
+
+#### Add Action
+
+When dynamically adding appointments to a data source, we must notify the data source for the add action.
+
+{% tabs %}
+{% highlight dart hl_lines="14 15 16" %}
+
+@override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: Scaffold(
+          floatingActionButton: FloatingActionButton(
+            child: const Text('Add appointment'),
+            onPressed: () {
+              final Appointment app = Appointment(
+                  startTime: _controller.displayDate!,
+                  endTime:
+                      _controller.displayDate!.add(const Duration(hours: 2)),
+                  subject: 'Add Appointment',
+                  color: Colors.pink);
+              _events?.appointments!.add(app);
+              _events?.notifyListeners(
+                  CalendarDataSourceAction.add, <Appointment>[app]);
+            },
+          ),
+          body: SfCalendar(
+            view: CalendarView.month,
+            controller: _controller,
+            dataSource: _events,
+          )),
+    );
+  }
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Remove Action
+
+When we remove appointments from a data source dynamically, we must notify the data source for the remove action.
+
+{% tabs %}
+{% highlight dart hl_lines="2 3 4" %}
+
+final Appointment removeAppointment = _events?.appointments![0];
```

Comment:
Highlight all lines

Reply:
I have checked and highlighted all the lines. I have updated my changes in the latest commit.